### PR TITLE
Track2 Management SDK for StackHCI

### PR DESCRIPTION
**Pull request by priyjain358**
# All SDK Contribution checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.
- [ ] **Please open PR in `Draft` mode if it is:**
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md).**

### [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code. (Track 2 only)
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK. Please double check nuget.org current release version.

## Additional management plane SDK specific contribution checklist: 
Note: Only applies to `Microsoft.Azure.Management.[RP]` or `Azure.ResourceManager.[RP]`
 
- [ ] Include updated [management metadata](https://github.com/Azure/azure-sdk-for-net/tree/main/eng/mgmt/mgmtmetadata).
- [ ] Update AzureRP.props to add/remove version info to maintain up to date API versions.

### Management plane SDK Troubleshooting
- If this is very first SDK for a services and you are adding new service folders directly under /SDK, please add `new service` label and/or contact assigned reviewer.
- If the check fails at the `Verify Code Generation` step, please ensure:
	- Do not modify any code in generated folders.
	- Do not selectively include/remove generated files in the PR.
	- Do use `generate.ps1/cmd` to generate this PR instead of calling `autorest` directly.
	Please pay attention to the @microsoft.csharp version output after running `generate.ps1`. If it is lower than current released version (2.3.82), please run it again as it should pull down the latest version.
	
	**Note: We have recently updated the PSH module called by `generate.ps1` to emit additional data. This would help reduce/eliminate the Code Verification check error. Please run following command**:

	    `dotnet msbuild eng/mgmt.proj /t:Util /p:UtilityName=InstallPsModules`

### Old outstanding PR cleanup
 Please note:
	If PRs (including draft) has been out for more than 60 days and there are no responses from our query or followups, they will be closed to maintain a concise list for our reviewers.

### CODE REVIEW

**Comment by fengzhou-msft** [comment]
Please add a `ci.yml`([example](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/compute/ci.yml)) file to trigger track2 validation.

**Comment by fengzhou-msft** [comment]
Please generate an api file by running `./eng/scripts/Export-API.ps1 azurestackhci`. We will mainly review the api from there for this PR.

File: sdk/azurestackhci/Azure.ResourceManager.StackHCI/src/Generated/Extension.cs [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+using Azure.Core.Pipeline;
+using Azure.ResourceManager;
+using Azure.ResourceManager.Core;
+using Azure.ResourceManager.StackHCI.Models;
+using Azure.ResourceManager.Resources.Models;
+
+namespace Azure.ResourceManager.StackHCI
+{
+    /// <summary> A Class representing a Extension along with the instance operations that can be performed on it. </summary>
+    public partial class Extension : ArmResource
```
Comment (fengzhou-msft) [inline sdk/azurestackhci/Azure.ResourceManager.StackHCI/src/Generated/Extension.cs:23]: Avoid using sing-word class names and use a more specific name with directive settings in autorest.md.

Comment (priyjain358) [reply sdk/azurestackhci/Azure.ResourceManager.StackHCI/src/Generated/Extension.cs:23]: This is an arm-resource present in the swagger. I'm not sure if we can rename this. I could only see directives worked for files under Models.

Comment (fengzhou-msft) [reply sdk/azurestackhci/Azure.ResourceManager.StackHCI/src/Generated/Extension.cs:23]: Use a directive like:
```
- from: filename.json
  where: $.definitions.Extension
  transform: $["x-ms-client-name"] = "XXXExtension"
```
Please also change for `Cluster`.

File: sdk/azurestackhci/Azure.ResourceManager.StackHCI/CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# Release History
+
+## 2.0.0-preview.1 (Unreleased)
```
Comment (fengzhou-msft) [inline sdk/azurestackhci/Azure.ResourceManager.StackHCI/CHANGELOG.md:3]: `preview` is deprecated, use `beta`.

File: sdk/azurestackhci/Azure.ResourceManager.StackHCI/README.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+This package follows the [new Azure SDK guidelines](https://azure.github.io/azure-sdk/general_introduction.html) which provide a number of core capabilities that are shared amongst all Azure SDKs, including the intuitive Azure Identity library, an HTTP Pipeline with custom policies, error-handling, distributed tracing, and much more.
+
+## Getting started 
+
+### Install the package
+
+Install the Azure Azure.ResourceManager.StackHCI management library for .NET with [NuGet](https://www.nuget.org/):
+
+```PowerShell
+Install-Package Azure.ResourceManager.StackHCI -Version 2.0.0-preview.1
```
Comment (fengzhou-msft) [inline sdk/azurestackhci/Azure.ResourceManager.StackHCI/README.md:12]: Update the beta version here as well.

Comment (priyjain358) [reply sdk/azurestackhci/Azure.ResourceManager.StackHCI/README.md]: Done

File: sdk/azurestackhci/Azure.ResourceManager.StackHCI/CHANGELOG.md (hunk 2 above)
Comment (priyjain358) [reply sdk/azurestackhci/Azure.ResourceManager.StackHCI/CHANGELOG.md]: done

File: sdk/azurestackhci/Azure.ResourceManager.StackHCI/src/Generated/Extension.cs (hunk 1 above)
Comment (priyjain358) [reply sdk/azurestackhci/Azure.ResourceManager.StackHCI/src/Generated/Extension.cs]: Done

File: sdk/azurestackhci/ci.yml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+trigger:
+  branches:
+    include:
+    - main
+    - hotfix/*
+    - release/*
+  paths:
+    include:
+    - sdk/azure.resourcemanager.stackhci/
```
Comment (fengzhou-msft) [inline sdk/azurestackhci/ci.yml:11]:
```suggestion
    - sdk/azurestackhci/
```

File: sdk/azurestackhci/ci.yml [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+pr:
+  branches:
+    include:
+    - main
+    - feature/*
+    - hotfix/*
+    - release/*
+  paths:
+    include:
+    - sdk/azure.resourcemanager.stackhci/
```
Comment (fengzhou-msft) [inline sdk/azurestackhci/ci.yml:22]:
```suggestion
    - sdk/azurestackhci/
```

File: sdk/azurestackhci/ci.yml [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+extends:
+  template: ../../eng/pipelines/templates/stages/archetype-sdk-client.yml
+  parameters:
+    ServiceDirectory: azure.resourcemanager.stackhci
```
Comment (fengzhou-msft) [inline sdk/azurestackhci/ci.yml:27]:
```suggestion
    ServiceDirectory: azurestackhci
```

**Review by fengzhou-msft — APPROVED** [review]

**Pull request merged**